### ISSUE TITLE
Feature: add ability to redefine hex value of 16 default terminal colors, and cursor -- this will enable having different terminal color themes in different panes

### ISSUE THREAD

**Issue by reportaman** [tmux/tmux#2815]
Being able to redefine the hex color value of 16 named colors ``black, red, green, yellow, blue, magenta, cyan, white`` and their bright counterparts ``brightred``, etc will truly enable having different color schemes in tmux panes. Currently tmux provides the ability to redefine only the hex value of background and foreground colors, while this is very useful, the range of values one can apply is limited by the parent terminals values for these named colors. Having this ability will truly enable having different terminal color themes in different panes.

Here's a screenshot of from my alacritty.yaml, ideally the STYLEs will include an ability to redefine hex color value of all of fields here (background+foreground is already implemented in tmux, redefining named colors and cursor colors are not yet supported)
![Screen Shot 2021-08-09 at 11 33 46 PM](https://user-images.githubusercontent.com/71736629/128820240-8e457f70-a177-4877-a6be-5459c3ceee97.png)

**Comment by nicm** [comment]
Applications can do this already by using OSC 4 to change the palette, although it is not exposed to the user. It would be easy enough to add an array option to for this.

**Comment by nicm** [comment]
Added to todo list.

**Comment by nicm** [comment]
Try this please which adds a window/pane option `palette`: [tmux-palette1.diff.txt](https://github.com/tmux/tmux/files/6967860/tmux-palette1.diff.txt)
~~~~
$ tput setaf 1 0 0
$ tmux set -p palette[1] '#0000ff'
$ tmux set -w palette[1] '#00ff00'
$ tmux set -up palette
$ tmux set -uw palette

**Comment by nicm** [comment]
Here is a better version that also applies the palette to popups (if the global `palette` option is set), and enables OSC 4, 104, 10, 110, 11 and 111 in popups: [tmux-palette2.diff.txt](https://github.com/tmux/tmux/files/6968428/tmux-palette2.diff.txt)

**Comment by nicm** [comment]
I have applied this now with a change so the option name is `pane-colours`, not `palette`.

**Comment by reportaman** [comment]
@nicm Yes I confirm it works well! I tried your patch ``tmux-palette2.diff.txt``.

In my testing, the named colors ``black, red, green, yellow, blue, magenta, cyan, white``, and their bright counterparts can be changed using index 0-15. Background and foreground can be changed using the usual STYLEs ``fg=colour``, ``bg=colour``. The only thing missing is a tmux way of setting different cursor color depending on pane/window. That will complete the set of colors one needs in a terminal color palette, and people can start writing tmux color-schemes.

I did come across ``ccolor` of ``terminal-overrides``, but that seems to be only a global option (perhaps set only once when tmux conf is loaded), and could not find the correct syntax to use it too.
